File: frontend/src/lib/types/lesson.ts
export interface Lesson {
    id: string;
    courseId: string;
    title: string;
    videoUrl: string | null;
    documentUrl: string | null;
    orderIndex: number;
    duration: number; // in minutes
    isPublished: boolean;
    createdAt: string;
    updatedAt: string;
}

export interface CreateLessonRequest {
    courseId: string;
    title: string;
    videoUrl?: string;
    documentUrl?: string;
    orderIndex?: number;
    duration?: number;
}

export interface UpdateLessonRequest {
    title?: string;
    videoUrl?: string;
    documentUrl?: string;
    orderIndex?: number;
    duration?: number;
    isPublished?: boolean;
}

File: backend/EduVN.Domain/Entities/Lesson.cs
namespace EduVN.Domain.Entities;

public class Lesson
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid CourseId { get; set; }
    public required string Title { get; set; }
    public string? VideoUrl { get; set; }
    public string? DocumentUrl { get; set; }
    public int OrderIndex { get; set; }
    public int Duration { get; set; } = 0; // In minutes
    public bool IsPublished { get; set; } = true;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Navigation properties
    public Course Course { get; set; } = null!;
}
